### DISCUSSION TITLE
Running Multiple Instances

### DISCUSSION THREAD

**Discussion by badger-mark** [spatie/file-system-watcher#27] [Q&A]
Looking to see if there is a better way to watch multiple directories and have different actions tied to them.  I am currently running the following but when testing the second directory doesn't get watched.  If I comment out the first watcher instance the second one works.

![image](https://github.com/spatie/file-system-watcher/assets/111913387/789b0433-310f-403d-b8a6-1399421bbeb6)

**Comment by nilskretschmer** [comment]
The issue is that the first watcher starts and the second watcher is never started, because the `start()`-function runs indefinitely.

Split it up into multiple commands or use the same watcher and separate the logic for the path (you get the path injected so you can separate your logic according to that path).
